### PR TITLE
Apply RetryOnConflict on Bulk documents

### PR DESCRIPTION
**Pull request by csabavirag**
This PR is created to solve the issue on the missing RetryOnConflict (retry_on_conflict) metadata attribute when adding documents in bulk (detailed at https://github.com/ruflin/Elastica/issues/2182)

### CODE REVIEW

**Review by ruflin — COMMENTED** [review]
Thanks for the PR. Change LGTM. Could you add an entry to the changelog? And can you check in the test suite if there is an easy way to add a test for this?

Normally we make changes to 8.x first and then backport to 7.x. But currently there is quite a large PR ongoing against 8.x to make it 8 compatible: https://github.com/ruflin/Elastica/pull/2181 What I propose is to get this in but in parallel follow up with a PR against the 8.x branch to ensure we don't forget about it. Otherwise I worry we will miss this feature in 8.x.

**Comment by ruflin** [comment]
@csabavirag I did just trigger the test suite and it seems to break one of the tests. It seems that in some cases the object is `null` so you need to add a check for this.

**Comment by csabavirag** [comment]
I pushed a new commit with does not set RetryOnConflict if it has already been set on the document/script. I know `retry_on_conflict` is stripped from the final message(s) if its value is `0`, but thought to exclude this case in order to a smaller memory footprint.
This allowed me to run BulkTest.php successfully. Only 1 error was reported, but I believe this is not related to these changes

Output of `make docker-run-phpunit PHPUNIT_OPTIONS="tests/BulkTest.php"`:
```
There was 1 failure:

1) Elastica\Test\BulkTest::testMemoryUsage
Failed asserting that 2.3990734932979025 is less than 1.31.
```

I also updated the `BulkTest` suite to check the parameter is set on the document/script properly when only the client has `retry_on_conflict` set ("global option").

File: CHANGELOG.md
Comment (ruflin) [inline CHANGELOG.md:11]: Should we put this under "fixed"? sounds more like a bug fix but also happy to keep it under Added.

**Review by ruflin — APPROVED** [review]
Change LGTM, waiting on CI to run through.

File: CHANGELOG.md
Comment (csabavirag) [reply CHANGELOG.md:11]: First I added to that section. Then I moved to Added. TBH, I would leave it up to you.

Comment (ruflin) [reply CHANGELOG.md:11]: Lets keep it, I can always clean it up later.

**Comment by ruflin** [comment]
@csabavirag Thank you for the contribution!

Could you follow up with a PR against 8.x. The big 8.x change made it in and we should make sure this change also makes it into 8.x

**Pull request merged**
